execution_engine/src/core/runtime/auction_internal.rs

```diff
@@ -215,6 +215,69 @@ where
         cl_value.into_t().map_err(|_| Error::CLValue)
     }
 
+    fn mint_into_existing_purse(
+        &mut self,
+        amount: U512,
+        existing_purse: URef,
+    ) -> Result<(), Error> {
+        if self.context.get_caller() != PublicKey::System.to_account_hash()
```

Comment:
Should we even allow for a non-system account to call `mint` endpoints? I don't think so.

Reply:
I think we should address that in a separate PR too if required - this PR matches the existing functionality I believe.